petpal/comments/views.py

```diff
@@ -4,61 +4,181 @@
+PERMISSION: User logged in and must be part of the application if is_review == False
 SUCCESS:
 '''
 @api_view(['POST'])
 def comment_create_view(request):
-    return Response({'msg': 'create comment'}, status=status.HTTP_200_OK)
+    if not request.user.is_authenticated:
+        return JsonResponse({'data': 'User not logged in'}, status=status.HTTP_401_UNAUTHORIZED)
+    is_author_seeker = request.data.get('is_author_seeker')
+    is_review = request.data.get('is_review')
+    content = request.data.get('content')
+    seeker_email = request.data.get('seeker_email')
+    shelter_email = request.data.get('shelter_email')
+    if is_review == None or is_author_seeker == None or content == None or seeker_email == None or shelter_email == None:
+        return JsonResponse({'data': 'Missing fields'}, status=status.HTTP_400_BAD_REQUEST)
+    if is_review == False:
+        application_id = request.data.get('application_id')
+        try:
+            application = Application.objects.get(pk=application_id)
+            # Check if user is part of the application
+            if request.user != application.seeker and request.user != application.petlisting.owner:
+                return JsonResponse({'data': 'User not authorized to comment on this application'}, status=status.HTTP_401_UNAUTHORIZED)
+            new_comment = Comment(content=content, is_author_seeker=is_author_seeker, seeker=application.seeker, shelter=application.petlisting.owner, is_review=is_review, application=application)
+            new_comment.save()
+            return JsonResponse({'data': 'Comment Created'}, status=status.HTTP_200_OK)
+        except:
+            return JsonResponse({'data': 'Application does not exist'}, status=status.HTTP_400_BAD_REQUEST)
+    else:
+        try:
+            seeker = User.objects.get(email=seeker_email)
+            shelter = User.objects.get(email=shelter_email)
+            rating = request.data.get('rating')
+            new_comment = Comment(content=content, is_author_seeker=is_author_seeker, seeker=seeker, shelter=shelter, is_review=is_review, rating=rating)
+            new_comment.save()
+            return JsonResponse({'data': 'Review Created'}, status=status.HTTP_200_OK)
+        except:
+            return JsonResponse({'data': 'User does not exist'}, status=status.HTTP_400_BAD_REQUEST)
 
 
 '''
 VIEW A Comment
 ENDPOINT: /api/comments/<str:msg_id>
 METHOD: GET
-PERMISSION:
+PERMISSION: User logged in and must be the author or recipient of the comment
 '''
 @api_view(['GET'])
 def comment_detail_view(request, msg_id):
-    return Response({'msg': 'view comment'}, status=status.HTTP_200_OK)
+    if not request.user.is_authenticated:
+        return JsonResponse({'data': 'User not logged in'}, status=status.HTTP_401_UNAUTHORIZED)
+    if request.user != comment.seeker and request.user != comment.shelter:
+        return JsonResponse({'data': 'User not authorized to view this comment'}, status=status.HTTP_401_UNAUTHORIZED)
```

Comment:
These 2 lines should be moved to inside the try block, after line 69, currently it runs into a comment references before assignment error.

Also is this really true? I thought everyone can see all the comments that are shelter reviews (not that I think we have any frontend pages that will support getting individual comments anyways but for just for P2 demo purposes)